src/components/People/AddPerson/AddPerson.tsx

```diff
@@ -0,0 +1,290 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Person } from '../../../types/Person';
+
+import './AddPerson.scss';
+
+type Props = {
+  people: Person[],
+  setPeople: (people: Person[]) => void,
+};
+
+export const AddPerson: React.FC<Props> = ({ people, setPeople }) => {
+  const navigate = useNavigate();
+
+  const returnToList = () => navigate('/people');
+
+  const [name, setName] = useState('');
+  const [gender, setGender] = useState('');
+  const [born, setBorn] = useState('');
+  const [died, setDied] = useState('');
+  const [mother, setMother] = useState('');
+  const [father, setFather] = useState('');
+
+  const errors: string[] = [];
+
+  if (!name.match(/[A-Z]{1}[a-z]{2,12}\s{1}[A-Z]{1}[a-z]{2,12}/g)) {
+    errors.push('name');
+  }
+
+  if (!gender) {
+    errors.push('gender');
+  }
+
+  if (!mother) {
+    errors.push('mother');
+  }
+
+  if (!father) {
+    errors.push('father');
+  }
+
+  if (born.length > 4 || +born < 1400) {
+    errors.push('born');
+  }
+
+  if (+died - +born > 150 || +died - +born <= 0) {
+    errors.push('died');
+  }
+
+  const addPerson = () => {
+    if (errors.length === 0) {
+      setPeople([...people, {
+        name,
+        sex: gender,
+        born: +born,
+        died: +died,
+        motherName: mother,
+        fatherName: father,
+        slug: `${name.replace(' ', '-').toLowerCase()}-${born}`,
+      }]);
+
+      returnToList();
+    }
+  };
+
+  return (
+    <div
+      className="addPerson"
+    >
+      <div className="addPerson__form">
+        <h1 className="addPerson__form--head">
+          Create New Entry
+        </h1>
+
+        <div className="addPerson__form--name">
+          <h3>
+            Please enter name
+            {!errors.includes('name') && <p className="OK OK--name">&#10004;</p>}
+          </h3>
+          <h6
+            className="form__error"
+            style={{
+              opacity: errors.includes('name') ? '1' : '0',
+            }}
```

Comment:
```suggestion
            style={{
              opacity: errors.includes('name') ? '1' : '0',
            }}
```
Для динамічної зміни стилів в реакті можна використовувати бібліотеку classnames
https://www.npmjs.com/package/classnames